src/value.rs

```diff
@@ -193,6 +194,29 @@ impl Value {
         )
     }
 
+    /// Encode a byte slice as a value.
+    ///
+    /// The length of the slice must be a power of two.
+    pub fn from_slice(v: &[u8]) -> Arc<Self> {
```

Comment:
I feel this can be mis-used. We cannot really parse a value from byte slice without knowing the exact type of value that we are parsing. For example, length of encoding for types `u16 * u1` is same as `u8 * u8` as it is important for caller to dis-ambiguate that. 

Do you want some form of convenience function that only operates on powers of two, and outputs (2^8^(2^n)) type? In this case, I would create a private function and re-name like something `from_pow_two_8_type`

Reply:
Also, I think the implementation can be something like `BitIter::read_value(powers_of_two(log(v.len() + 3)))`.

Reply:
You're right; `from_slice` is ambiguous. I renamed the method to `power_of_two`, similar to `Value::u{1,2,4,8,16,32,64,128}`.

I don't want to use `BitIter` because `power_of_two` is supposed to replace `u{256,512}_from_slice`.